staging/mausezahn.c

```diff
@@ -676,11 +676,33 @@ int getopts (int argc, char *argv[])
 				(!ipv6_mode && get_ip_range_src(tx.ip_src_txt))
 				)
 			{
-				// name2addr4 accepts a DOTTED DECIMAL ADDRESS or a FQDN:
-				if (ipv6_mode)
+				// name2addr{4,6} accept a valid IP address or a FQDN:
+				if (ipv6_mode) {
 					tx.ip6_src = libnet_name2addr6 (l, tx.ip_src_txt, LIBNET_RESOLVE);
-				else
-					tx.ip_src = libnet_name2addr4 (l, tx.ip_src_txt, LIBNET_RESOLVE);
+					struct in6_addr src_check;
+					if (
+					(strncmp((char*)&tx.ip6_src,(char*)&in6addr_error,sizeof(in6addr_error)) == 0) &&
+					// additional inet_pton() check to cover case where ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff
+					// is specified as source
```

Comment:
Agreed, please use /* */ comments. Also the if clause is line-wrapped a bit weird. And as @vkochan suggested, pleas use `libnet_in6_is_error()` if applicable.